### ISSUE TITLE
Some of the Flow types seem incorrect

### ISSUE THREAD

**Issue by echo99** [respond-framework/rudy#30]
I've been trying to take a stab at attempting to use `rudy`, but I've run into a few inconsistencies between the defined Flow types and the actual parameters that get passed to the thunks.

I see that the thunks are defined as `StandardCallback` which has the following definition:
```javascript
export type StandardCallback = (
  dispatch: Dispatch,
  getState: GetState,
  bag: Bag,
) => ?any | Promise<any>
```
but when using the library the `dispatch` parameter gets passed a `Request` object. Same deal with `Options.beforeLeave`, which is expecting a state object.

Totally understand if this library isn't yet ready for general consumption but thought I'd bring it up. :slightly_smiling_face:

**Comment by hedgepigdaniel** [comment]
Yes - there is significant work to be done to update the flow types (many of them haven't been updated since the API changed from redux-first-router). I've been going through different components and bringing them up to date, but there is plenty left.

In this case, the route and global callbacks are passed a single parameter, the request object. `dispatch` and `getState` are properties of the request.

TL;DR if you're coming from RFR, you can change `(dispatch, getState, bag) => {}` to `({dispatch, getState, ...bagAndOtherStuff}) => {}`

**Comment by hedgepigdaniel** [comment]
Closing as the flow types are no longer exported. Future development will focus on improving the typescript types, which are incomplete, but accurate.
